FrontEnd/src/components/profileList/ProfileListPage.jsx

```diff
@@ -0,0 +1,128 @@
+import css from "./ProfileListPage.module.css";
+import ProfileList from "./ProfileList";
+import useSWR from "swr";
+import { useParams } from "react-router-dom";
+import { useState } from "react";
+import { ConfigProvider, Radio } from "antd";
+import Loader from "./Loader";
+import ErrorPage from "./ErrorPage";
+
+
+export default function ProfileListPage(props) {
+  let isAuthorized = props.isAuthorized;
+
+  const { filter } = useParams();
+
+  let profileTypeFilter = "";
+
+  let profileActivityFilter = "activities__name=";
+  const emptyActivityFilter = profileActivityFilter.length;
+
+  const [filterSaved, setFilterSaved] = useState(false);
+  const [currentPage, setCurrentPage] = useState(1);
+
+  switch (filter) {
+    case "companies":
+      profileTypeFilter = "is_registered=True";
+      break;
+    case "startups":
+      profileTypeFilter = "is_startup=True";
+      break;
+    case "producers":
+      profileActivityFilter += "Виробник";
+      break;
+    case "importers":
+      profileActivityFilter += "Імпортер";
+      break;
+    case "retailers":
+      profileActivityFilter += "Роздрібна мережа";
+      break;
+    case "horeca":
+      profileActivityFilter += "HORECA";
+      break;
+    default:
+      break;
+  }
+
+  const urlForAll = `${
+    process.env.REACT_APP_BASE_API_URL
+  }/api/profiles/?${profileTypeFilter}${
+    profileActivityFilter.length > emptyActivityFilter
+      ? profileActivityFilter
+      : ""
+  }&page=${currentPage}`;
+
+  const urlForSaved = `${
+    process.env.REACT_APP_BASE_API_URL
+  }/api/profiles/?${profileTypeFilter}${
+    profileActivityFilter.length > emptyActivityFilter
+      ? profileActivityFilter
+      : ""
+  }${filterSaved ? "&is_saved=True" : ""}&page=${currentPage}`;
+
+  const fetcher = (...args) => fetch(...args).then((res) => res.json());
+  const {
+    data: fetchedProfiles,
+    error,
+    isLoading,
+  } = useSWR(filterSaved ? urlForSaved : urlForAll, fetcher);
+  // const { trigger, isMutating } = useSWRMutation(urlForAll, fetcher)
```

Comment:
Do we use mutation on this page?

Reply:
Nope. This is a leftover from my previous tries with mutations. Removed